server/src/saros/server/session/NegotiationHandler.java

```diff
@@ -146,15 +136,62 @@ public void run() {
   public void handleIncomingProjectNegotiation(
       final AbstractIncomingProjectNegotiation negotiation) {
 
+    Map<String, IProject> projectMapping = new HashMap<>();
+
+    for (ProjectNegotiationData data : negotiation.getProjectNegotiationData()) {
+      String projectName = data.getProjectName();
+      IProject project = workspace.getProject(projectName);
+
+      // TODO: The file path is currently dictated by the name, potentially resulting in CONFLICTS
+      if (!project.exists()) {
+        try {
+          project.adaptTo(ServerProjectImpl.class).create();
+        } catch (IOException e) {
+          projectMapping.clear();
+          negotiation.localCancel(
+              "Error creating project folder", NegotiationTools.CancelOption.NOTIFY_PEER);
+          break;
```

Comment:
Is a break really the right statement in this case? Do you want to still add the project to the mapping and continue with starting the session negotiation, even though the project creation failed (and you canceled the negotiation)?

Reply:
Sorry, misread where the loop ended. So the part of the project mapping does not apply. But I am still not sure whether it is sensible to call the `projectExecutor` to run the negotiation, even though you already canceled it locally.

Reply:
Yeah, you are right